lib/core/src/configuration_hook_manager.cpp

```diff
@@ -0,0 +1,44 @@
+#include "irods/configuration_hook_manager.hpp"
+#include "irods/irods_logger.hpp"
+
+#include <unordered_map>
+
+namespace irods::experimental
+{
+    namespace
+    {
+        std::vector<hook> hooks;
+        std::unordered_multimap<std::string, size_t> edge_sets;
```

Comment:
Please document these variables.

These don't require doxygen style comments.

Make sure to explain any details needed for proper maintenance of this library. Without good explanations of these variables, someone will be tempted to re-implement this 6 to 12 months from now.